src/echo/echo.rs

```diff
@@ -87,84 +68,75 @@ fn convert_str(string: &[u8], index: usize, base: u32) -> (char, usize) {
                     } else {
-                        prev_was_slash = true;
+                        true
                     }
                 } else {
-                    prev_was_slash = false;
                     match c {
                         '\\' => print!("\\"),
+                        'n' => print!("\n"),
+                        'r' => print!("\r"),
+                        't' => print!("\t"),
+                        'v' => print!("\x0B"),
                         'a' => print!("\x07"),
                         'b' => print!("\x08"),
                         'c' => break,
                         'e' => print!("\x1B"),
                         'f' => print!("\x0C"),
-                        'n' => print!("\n"),
-                        'r' => print!("\r"),
-                        't' => print!("\t"),
-                        'v' => print!("\x0B"),
-                        'x' => {
-                            let (c, num_char_used) = convert_str(string.as_bytes(), index + 1, 16);
-                            if num_char_used == 0 {
-                                print!("\\x");
-                            } else {
-                                print!("{}", c);
-                                for _ in 0 .. num_char_used {
-                                    iter.next(); // consume used characters
-                                }
-                            }
-                        },
-                        '0' => {
-                            let (c, num_char_used) = convert_str(string.as_bytes(), index + 1, 8);
-                            if num_char_used == 0 {
-                                print!("\0");
-                            } else {
-                                print!("{}", c);
-                                for _ in 0 .. num_char_used {
-                                    iter.next(); // consume used characters
-                                }
-                            }
-                        }
-                        _ => {
-                            let (esc_c, num_char_used) = convert_str(string.as_bytes(), index, 8);
-                            if num_char_used == 0 {
-                                print!("\\{}", c);
-                            } else {
-                                print!("{}", esc_c);
-                                for _ in 1 .. num_char_used {
-                                    iter.next(); // consume used characters
+                        ch => { // 'x' or '0' or _
+                            match convert_str(string.as_bytes(),
+                                              if ch == 'x' || ch == '0' {
+                                                  idx + 1
+                                              } else {
+                                                  idx
+                                              },
+                                              if ch == 'x' { Base::B16 } else { Base::B8 }) {
```

Comment:
The second and third arguments of this function call are kind of hard to read (because it's all in a `match` expression).  I'd prefer if you just assign them to variables and pass those to the function.